hand-eye-calibration/full_hand_eye_sample.py

```diff
@@ -0,0 +1,510 @@
+        required=True,
+    )
+    parser.add_argument("--ip", help="IP address to robot", required=True)
+
+    return parser.parse_args()
+
+
+def _write_robot_state(
+    con: rtde, input_data, finish_capture: bool = False, camera_ready: bool = False
+):
+    """Write to robot I/O registrer
+
+    Args:
+        con: Connection between computer and robot
+        input_data: Input package containing the specific input data registers
+        finish_capture: Boolean value to robot_state that q_r scene capture is finished
+        camera_ready: Boolean value to robot_state that camera is ready to capture images
+    """
+
+    if finish_capture:
+        input_data.input_bit_register_64 = 1
+    else:
+        input_data.input_bit_register_64 = 0
+
+    if camera_ready:
+        input_data.input_bit_register_65 = 1
+    else:
+        input_data.input_bit_register_65 = 0
+
+    con.send(input_data)
+
+
+def _initialize_robot_sync(host: str, port: int):
+    """Set up communication with UR robot
+
+    Args:
+        host: IP address
+        port: Port number
+
+    Returns:
+        Connection to robot
+        Package containing the specific input data registers
+
+    Raises:
+        RuntimeError: If computer is not able to establish communication with robot
+    """
+
+    conf = rtde_config.ConfigFile(Path(Path.cwd() / "robot_communication_file.xml"))
+    output_names, output_types = conf.get_recipe("out")
+    input_names, input_types = conf.get_recipe("in")
+
+    con = rtde.RTDE(host, port)
+    con.connect()
+
+    # To ensure that the application is compatable with further versions of UR controller
+    if not con.negotiate_protocol_version():
+        raise RuntimeError(f"Protocol do not match")
+
+    if not con.send_output_setup(output_names, output_types, frequency=200):
+        raise RuntimeError(f"Unable to configure output")
+
+    robot_input_data = con.send_input_setup(input_names, input_types)
+
+    if not con.send_start():
+        raise RuntimeError(f"Unable to start synchronization")
+
+    print("Communication initialization completed. \n")
+
+    return con, robot_input_data
+
+
+def _save_zdf_and_pose(
+    save_dir: Path, image_num: int, frame: zivid.Frame, transform: np.array
+):
+    """Save data to folder
+
+    Args:
+        save_dir: Directory to save data
+        image_num: Image number
+        frame: Point cloud stored as .zdf
+        transform: 4x4 transformation matrix
+    """
+
+    frame.save(save_dir / f"img{image_num:02d}.zdf")
+
+    file_storage = cv2.FileStorage(
+        str(save_dir / f"pos{image_num:02d}.yaml"), cv2.FILE_STORAGE_WRITE
+    )
+    file_storage.write("PoseState", transform)
+    file_storage.release()
+
+
+def _generate_folder():
+    """Generate folder where dataset weill be stored
+
+    Returns:
+        Directory to where data will be saved
+    """
+
+    location_dir = Path(
+        Path.cwd() / "datasets" / datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+    )
```

Comment:
I don't think you need to wrap this in `Path()` constructor. "Path / string" already returns a Path object.

Reply:
done